include/pmacc/random/distributions/uniform/Uniform_double.hpp

```diff
@@ -0,0 +1,177 @@
+/* Copyright 2015-2018 Alexander Grund, Rene Widera
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc/types.hpp"
+#include "pmacc/random/distributions/Uniform.hpp"
+#include "pmacc/random/distributions/uniform/Range.hpp"
+
+namespace pmacc
+{
+namespace random
+{
+namespace distributions
+{
+namespace detail
+{
+
+    /** Returns a random double value uniformly distributed in (0,1]
+     *
+     * The smallest created value is `2^-65` (~ `2.710505431213761*10^-20`)
+     */
+    template<class T_RNGMethod>
+    class Uniform<
+        uniform::ExcludeZero<double>,
+        T_RNGMethod,
+        void
+    >
+    {
+    public:
+        typedef T_RNGMethod RNGMethod;
+        typedef typename RNGMethod::StateType StateType;
+        typedef double result_type;
+
+        template< typename T_Acc >
+        DINLINE double
+        operator()(
+            T_Acc const & acc,
+            StateType& state
+        ) const
+        {
+            double const value2pow64Inv = 5.421010862427522e-20;
+            uint64_t const random = RNGMethod().get64Bits(
+                acc,
+                state
+            );
+            return static_cast< double >( random ) * value2pow64Inv +
```

Comment:
ANSWER to previous questions and NEW QUESTION:
It is quasi-uniform in the real numbers **but** what is the result of `0.0 + ( value2pow64Inv / 2.0 )` - `0` or `value2pow64Inv`? It should be `0.0`.

Reply:
Okay - all fine - this is **Exclude Zero**.